Stratis.Bitcoin/Features/Wallet/WalletManager.cs

```diff
@@ -413,7 +421,8 @@ public int LastBlockHeight()
 				return this.chain.Tip.Height;
 			}
 
-			return this.Wallets.Min(w => w.AccountsRoot.Single(a => a.CoinType == this.coinType).LastBlockSyncedHeight) ?? 0;
+            // todo: use max instead of min?
```

Comment:
I see where the confusion comes from.
It's true that it's called "LastBlockHeight" but in fact this doesn't describe the last block height that was indeed received but rather the height of the block from which we want to sync.
So imagine that you have 2 wallets, one synced up to block 500 and the other one up to block 800. Although the last block received is 800, we're going to behave as if the last block received is 500 because we need to sync from there.
Does that make sense?